### PR TITLE
Fix dev setup

### LINKED ISSUE #222: Fix dev setup issues
Fix some issues with setting up nplinker development on Linux

- [x] Fix install-nplinker-deps quitting early
  - [x] Add sudo to apt calls
- [x] Fix version of pytest to be <8
  - This addresses a problem with pytest_lazyfixture (https://github.com/pytest-dev/pytest/issues/11890)

### PR DESCRIPTION
**Pull request by adraismawur**
Closes #222 

This fixes some minor issues I had setting up the dev environment on my Linux machine.

Changes:
- Adds sudo to the apt commands used under Linux
- Fixes the version of pytest-fixtures to be <8 to resolve an AttributeError

### CODE REVIEW

File: bin/install-nplinker-deps [hunk 1]
```diff
@@ -57,8 +57,8 @@ if [[ "$OSTYPE" == "linux"* ]]; then
         if ${run_on_github}; then
             do_install="sudo apt install -y"
         else
-            apt update &> /dev/null
-            do_install="apt install -y"
+            sudo apt update &> /dev/null
```
Comment (CunliangGeng) [inline bin/install-nplinker-deps:61]:
```suggestion
            echo "To use 'apt' command, you need to allow 'sudo apt' by providing your computer password"
            sudo apt update &> /dev/null
```

File: requirements.dev.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ autoflake
 build
 bump2version
 coverage[toml]
-pytest
+pytest<8
```
Comment (CunliangGeng) [inline requirements.dev.txt]: lazy fixture is not used in the codebase, so it's better to remove the dependency of `pytest-lazy-fixture` from `requirements.txt`.

**Review by CunliangGeng — CHANGES_REQUESTED** [review]
Thanks for the update, I left two minor comments.

**Comment by CunliangGeng** [comment]
The github actions are not passed, which are caused by unit tests (worked well locally), I'll take a look into that. You can ignore the actions problem for this PR.

File: requirements.dev.txt (hunk 2 above)
Comment (adraismawur) [reply requirements.dev.txt]: Ok! Done and the pytest dependency is back to latest version.

**Review by CunliangGeng — APPROVED** [review]
Great! It's approved and you can merge it now.

**Comment by adraismawur** [comment]
Thanks! But it seems I do not have write access to the repository & cannot merge.

**Comment by CunliangGeng** [comment]
> Thanks! But it seems I do not have write access to the repository & cannot merge.

Aha, I only added you to the org and forgot to give you write access to the repo. Just gave you the access, check please.

**Pull request merged**